marketpulse/main/models.py

```diff
@@ -0,0 +1,50 @@
+from django.contrib.auth.models import User
+from django.db import models
+
+from marketpulse.devices.models import Device
+from marketpulse.geo.models import LocaltionBase
+
+
+class Activity(models.Model):
+    name = models.CharField(max_length=100)
+    slug = models.SlugField(blank=True, max_length=255)
+
+    class Meta:
+        ordering = ['name']
+        verbose_name = 'activity'
+        verbose_name_plural = 'activities'
+
+    def __unicode__(self):
+        return self.name
+
+
+class Location(LocaltionBase):
+    address = models.CharField(max_length=120)
+    shop_name = models.CharField(max_length=120, default='', blank=True)
+    link = models.URLField(max_length=500, blank=True, default='')
+
+
+class Price(models.Model):
+    amount = models.IntegerField()
+    currency = models.CharField(max_length=3, choices=[])
+
+
+class Plan(models.Model):
+    duration = models.IntegerField()
+    description = models.TextField(default='', blank=True)
+    price = models.ForeignKey(Price, related_name='plans')
+
+
+class Contribution(models.Model):
+    """Model for contribution data."""
+
+    user = models.ForeignKey(User, related_name='contributions')
+    activity = models.ForeignKey(Activity, related_name='contributions')
+    location = models.ForeignKey(Location, related_name='contributions')
+    device = models.ForeignKey(Device, related_name='contributions')
+    created_on = models.DateTimeField(auto_now_add=True)
+    updated_on = models.DateTimeField(auto_now=True)
+    comment = models.TextField(blank=True, default='')
+    availability = models.BooleanField(default=True)
+    price = models.ForeignKey(Price, null=True, default=None)
+    plan = models.ForeignKey(Plan, null=True, default=None)
```

Comment:
We may have multiple plans per contribution, so this should probably be manytomany.